### ISSUE TITLE
alasql.fn.total is not a function

### ISSUE THREAD

**Issue by mathiasrw** [AlaSQL/alasql#1530]
_Spawn from https://github.com/AlaSQL/alasql-sqllogictest/blob/a10ff82e75edafae2ce96c9d02ab47cfc85d4507/results/output.md#3620-testevidenceslt_lang_aggfunctest_

    SELECT total(y) FROM t1
    
    alasql.fn.total is not a function

We would like to support TOTAL. Its almost the same as SUM but 

> This function is similar to the [sum()](https://database.guide/how-sqlite-sum-works/) function, [except in the way it handles NULL inputs](https://database.guide/sqlite-sum-vs-total-whats-the-difference/). When there are no non-NULL values, then sum() returns NULL (instead of 0.0 like the total() function returns).

https://database.guide/how-sqlite-total-works/

**Comment by gangadhargo** [comment]
@mathiasrw I can pick this up if you verify and approve the PR for #1537  as I handled the SUM function in the PR which is almost similar to TOTAL.  :)

**Comment by mathiasrw** [comment]
Amazing. Thank you!

See if you can reuse some of the code so we dont duplicate everything. Also consider if it makes sense to use a function that you can call to avoid repeating `typeof == 'number' && typeof ==.....` 

Another input: the codebase was created before the tick notation was supported. I propose that you look into using something like 

```
var pre = 'if(
						typeof ${colexp} !="undefined" 
 						&& (!g[\'$$_VALUES_${colas}'][${colexp}])
					){`;

```

instead of 

```
var pre =
							'if(typeof ' +
							colexp +
							'!="undefined" && (!g[\'$$_VALUES_' +
							colas +
							"'][" +
							colexp +
							'])) {';
```

as it makes everything much easier to read.

**Comment by gangadhargo** [comment]
@mathiasrw what should be the expected behavior if strings or booleans or nulls exist while calculating the total? Is my assumption correct for the following case? 
var data = [
			{a: null, b: 9, c: true, c2: 1, d: null, e: 'XYZ1', f: new Number(2)},
			{a: null, b: 1, c: false, c2: false, d: 5, e: 'XYZ2', f: new Number(11)},
		];
  expected: {a: 0, b: 10, c: 0, c2: 1, d: 5, e: 0, f: 13}

**Comment by mathiasrw** [comment]
That is such a good question. 

Hmmm. Let me check the other DBs.

**Comment by mathiasrw** [comment]
I propose aggregating the values if they can be translated natively to a number. 

A bit like this 

    sum = sum + (val == +val ? +val:0)

This means the following result:

```
var data = [
			{a: null, b: 9, c: true, c2: 1, d: null, e: 'XYZ1', f: new Number(2)},
			{a: null, b: 1, c: false, c2: false, d: 5, e: 'XYZ2', f: new Number(11)},
		];

expected: {a: 0, b: 10, c: 1, c2: 1, d: 5, e: 0, f: 13}
```

Notice how true = 1 so c is different than your proposal

**Comment by gangadhargo** [comment]
the sum of numbers in strings like '+44' and '-43' = 1.  shouldn't we cover this case in aggregator SUM as well? or it is a specific case of aggregator TOTAL only?

**Comment by mathiasrw** [comment]
We could expand SUM if that makes it easier regarding the implementataion - but its a breaking change, so lets only do it if that makes your fix easier

**Comment by mathiasrw** [comment]
Released as part of v2.5.0
